DataService.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  DataService.swift
+//  DevAcademy
+//
+//  Created by Zdenko Čepan on 18/07/2023.
+//
+
+import Foundation
+
+enum MyError: Error {
+    case noData
+}
+
+
+class DataService {
+    var data: Result<Features, Error>?
+    
+    static let shared = DataService.init()
+    
+    private init() {
+        
+    }
+    
+    func fetchData(_ n: @escaping (Result<Features, Error>) -> Void) {
+        guard data == nil else {
+            n(Result.failure(MyError.noData))
+            return
+        }
```

Comment:
⚠️ 
Logika by měla být opačně:
 - pokud je `data == nil`, potom pokračujeme dále
 - pokud `data` něco obsahují, zavoláme closure ihned jako `success`

Reply:
Tady za to zmateni muzu nejspis ja, ze jsem nemel poradne prostudovane zadani. 😅 

Zacali jsme implementovat kontrolu dat, ktere jsem myslel ze mame vratit, ale ty data se meli teprve priradit. A pak uz jsme se k tomu nevratili.

@ZdenkoKrio  zkus to prosim upravit podle toho jak pise @mikolasstuchlik, a omlouvam se za zmateni.

Reply:
otázka k weak reference čo sa tam použila, nebola by v tomto prípade vhodnejšia unowned reference? (respektíve nesplňa dany prípad aj podmineky pre použitie unowned reference)

referencí dostává timer, ktorý existuje v metóde "singletonu" a teda jedinu silnu referenciu má class sama na seba.
podla nákresu a vysvetlenia z dokomentácie o ARC mi príde, že by tam mohola byť aj unowned reference

Reply:
Dobra otazka, premyslim 🤔 ...obecne jsem s tema `unowned` referencema opatrnej, protoze lepsi provolat nil nez skoncit crashem. 😅 ...Ale rekl bych, ze tim, ze se jedna o singleton (a mame tam to `static let`), tak realne nemuze dojit k uvolneni te DataService (a zadnou jinou instanci taky neudelame), takze v tomhle pripade by se ten `unowned` dal pouzit. 🤔 

Klidne me @mikolasstuchlik oprav nebo dopln.